presto-main/src/test/java/com/facebook/presto/operator/aggregation/differentialentropy/TestFixedHistogramAggregation.java

```diff
@@ -0,0 +1,250 @@
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("0.2"));
+        }
+    }
+
+    @Test
+    public void testIllegalMinMax()
+    {
+        try {
+            assertAggregation(
+                    this.function,
+                    0.0,
+                    createLongsBlock(Long.valueOf(200)),
+                    createDoublesBlock(Double.valueOf(0.1)),
+                    createDoublesBlock(Double.valueOf(0.2)),
+                    createStringsBlock(this.method),
+                    createDoublesBlock(Double.valueOf(0.2)),
+                    createDoublesBlock(Double.valueOf(0.1)));
+            fail("Expected PrestoException");
+        }
+        catch (PrestoException e) {
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("0.1"));
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("0.2"));
+        }
+    }
+
+    @Override
+    public Block[] getSequenceBlocks(int start, int length)
+    {
+        int sequenceLength = 2 * length;
+        BlockBuilder samplesBuilder = DOUBLE.createBlockBuilder(null, sequenceLength);
+        BlockBuilder weightsBuilder = DOUBLE.createBlockBuilder(null, sequenceLength);
+        BlockBuilder methodBuilder = VARCHAR.createBlockBuilder(null, sequenceLength);
+        for (int weight = 1; weight < 3; ++weight) {
+            for (int i = start; i < start + length; i++) {
+                final int bin = Math.max(Math.min(i, TestFixedHistogramAggregation.NUM_BINS - 1), 0);
+                DOUBLE.writeDouble(samplesBuilder, Double.valueOf(bin));
+                DOUBLE.writeDouble(weightsBuilder, Double.valueOf(weight));
+                VARCHAR.writeSlice(methodBuilder, wrappedBuffer(this.method.getBytes()));
+            }
+        }
+
+        return new Block[] {
+                createRLEBlock(NUM_BINS, sequenceLength),
+                samplesBuilder.build(),
+                weightsBuilder.build(),
+                methodBuilder.build(),
+                createRLEBlock(0.0, sequenceLength),
+                createRLEBlock((double) NUM_BINS, sequenceLength)
+        };
+    }
+
+    @Override
+    protected String getFunctionName()
+    {
+        return "differential_entropy";
+    }
+
+    @Override
+    protected List<String> getFunctionParameterTypes()
+    {
+        return ImmutableList.of(
+                StandardTypes.INTEGER,
+                StandardTypes.DOUBLE,
+                StandardTypes.DOUBLE,
+                StandardTypes.VARCHAR,
+                StandardTypes.DOUBLE,
+                StandardTypes.DOUBLE);
+    }
+
+    protected static void generateSamplesAndWeights(int start, int length, List<Double> samples, List<Double> weights)
+    {
+        for (int weight = 1; weight < 3; ++weight) {
+            for (int i = start; i < start + length; ++i) {
+                int bin = Math.max(Math.min(i, NUM_BINS - 1), 0);
```

Comment:
- inline `bin` variable 
- replace Double.valueOf with cast to double
- ++weight -> weight++

```
    protected static void generateSamplesAndWeights(int start, int length, List<Double> samples, List<Double> weights)
    {
        for (int weight = 1; weight < 3; weight++) {
            for (int i = start; i < start + length; ++i) {
                samples.add((double) Math.max(Math.min(i, NUM_BINS - 1), 0));
                weights.add((double) weight);
            }
        }
    }
```